scipy/stats/_binned_statistic.py

```diff
@@ -605,19 +596,13 @@ def binned_statistic_dd(sample, values, statistic='mean',
             result[vv, a] = flatsum
     elif statistic == 'median':
         result.fill(np.nan)
-        for i in np.unique(binnumbers):
```

Comment:
Just a side comment, looking at this now, it makes it faster ***and*** more readable (IMO). The PR removes some lines of code that are replicated boilerplate.